libbeat/publisher/queue/memqueue/runloop.go

```diff
@@ -82,10 +98,19 @@ func newRunLoop(broker *broker, observer queue.Observer) *runLoop {
 			<-timer.C
 		}
 	}
+
+	eventBufSize := broker.settings.Events
+	if broker.useByteLimits() {
+		// The queue is using byte limits, start with a buffer of 2^10 and
+		// we will expand it as needed.
+		eventBufSize = 1 << 10
```

Comment:
If your memory no longer has space for the powers of two, this reads bigger than it is. Why not just use 1024 directly?